### PR TITLE
refactor: base element for datepicker buttons

### LINKED ISSUE #2531: datepicker-previous-day and datepicker-next-day should share common logic (mixin or base class)

### PR DESCRIPTION
**Pull request by DavideMininni-Fincons**
## Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/lyne-design-system/lyne-components/blob/main/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/lyne-design-system/lyne-components/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [pull request tracker](https://github.com/lyne-design-system/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

This PR Closes #2531 

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../REVIEW.md) for more information what is checked during review process.

## Changes

Changes in this pull request:

- a new base class named `SbbDatepickerButton` has been created to include all the shared code between the `SbbDatepickerPreviousDay` and the `SbbDatepickerNextDay`

## Browsers

I tested the build on the following browsers:

- [ ] Firefox Desktop
- [x] Chrome Desktop
- [ ] Edge Desktop
- [ ] Safari Desktop
- [ ] Chrome Mobile
- [ ] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [ ] JAWS Firefox Desktop
- [ ] JAWS Chrome Desktop
- [ ] NVDA Firefox Desktop
- [ ] NVDA Chrome Desktop
- [ ] VoiceOver Safari Desktop
- [ ] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/components/datepicker/common/datepicker-button.ts [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      this.setAttribute('aria-disabled', 'true');
+      this.removeAttribute('tabindex');
+    } else {
+      this.removeAttribute('aria-disabled');
+      this.setAttribute('tabindex', '0');
+    }
+  }
+
+  protected override renderTemplate(): TemplateResult {
+    this._setDisabledRenderAttributes();
```
Comment (jeripeierSBB) [inline src/components/datepicker/common/datepicker-button.ts:198]: Maybe use the moment and set the attributes when needed and not in every rendering?

File: src/components/datepicker/common/datepicker-button.ts [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        ? i18nToday[this._language.current].toLowerCase()
+        : this._dateAdapter.getAccessibilityFormatDate(currentDate);
+
+    this.setAttribute(
+      'aria-label',
+      this.i18nSelectOffBoundaryDay(currentDateString)[this._language.current],
+    );
+  }
+
+  private _setDisabledRenderAttributes(): void {
```
Comment (jeripeierSBB) [inline src/components/datepicker/common/datepicker-button.ts:186]: would the disabled action mixin help here?

**Review by jeripeierSBB — APPROVED** [review]
LGTM, a possible improvement in a comment

File: src/components/datepicker/common/datepicker-button.ts (hunk 1 above)
Comment (DavideMininni-Fincons) [reply src/components/datepicker/common/datepicker-button.ts:198]: Do you feel should i move the method in the `willUpdate` callback?

File: src/components/datepicker/common/datepicker-button.ts (hunk 2 above)
Comment (DavideMininni-Fincons) [reply src/components/datepicker/common/datepicker-button.ts:186]: I think not, because the component would inherit an unwanted `disabled` prop

Comment (jeripeierSBB) [reply src/components/datepicker/common/datepicker-button.ts:183]: OK I see :)

File: src/components/datepicker/common/datepicker-button.ts (hunk 1 above)
Comment (DavideMininni-Fincons) [reply src/components/datepicker/common/datepicker-button.ts]: I removed it and called the method when the two variables are changed

**Pull request merged**